src/Text/Pandoc/Shared.hs

```diff
@@ -779,6 +780,31 @@ filterIpynbOutput mode = walk go
                     | otherwise = ""
         go x = x
 
+-- | Move formatting within code inlines to the outside,
+-- i.e. `Strong [ Code ("",[],[]) "Strong code" ]` (issue #7525)
+formatCode :: Inlines -> Inlines
+formatCode =
+  B.fromList
+    . concatMap fmt
+    . groupBy (\a b -> isPlaintext a && isPlaintext b)
+    . B.toList
+  where
+    isPlaintext (Str _) = True
+    isPlaintext Space = True
+    isPlaintext SoftBreak = True
+    isPlaintext (Quoted _ _) = True
+    isPlaintext _ = False
+    fmt xs
+        | all isPlaintext xs = B.toList $ B.code $ stringify xs
+        | any isPlaintext xs = B.toList $ formatCode $ B.fromList xs
+        | otherwise = walk
+                          ( \ys ->
+                              if length ys == 1 && not (isPlaintext $ head xs)
```

Comment:
I'd like to avoid partial functions like `head` (even though in this case you're checking length). I think maybe you could do something more elegant like

```
\case
  [y] | not (isPlaintext y) ->
  ys ->
```